chain-api/src/utils/signatures.spec.ts

```diff
@@ -315,4 +315,20 @@ describe("signatures", () => {
       s: new BN("14d3aed3bf7e07cb3bf2ef2c06cfde6db461eea8f58827df5b0fa4185d6535", "hex")
     });
   });
+
+  it("test metamask signatures vs galachain signatures", async () => {
+    const metamask = "5078520b05186d8babacee43d061f14b3575ad2999e561772b57032aa019bc2a7b01eb5ec412c9330d343025697e9449a0766995e3646941948e4acf0d0dff501c"
+    const metamaskPayload = {"quantity":"1","to":"client|63580d94c574ad78b121c267","tokenInstance":{"additionalKey":"none","category":"Unit","collection":"GALA","instance":"0","type":"none"},"uniqueKey":"26d4122e-34c8-4639-baa6-4382b398e68e"}
+
+    const galachain = "4ae122398fb2e69f95d7322043d72d18fce83a0a034c8faa5643d673693ae0c2a6ccb049fdff8d2220015f20635f6cb888fec60df2c3ae5eb1e5b6e0e8785cf21c"
+    const galachainPayload = {"quantity":"1","to":"client|63580d94c574ad78b121c267","tokenInstance":{"additionalKey":"none","category":"Unit","collection":"GALA","instance":"0","type":"none"},"uniqueKey":"8ad19f56-453a-40c8-aee5-11c0f753c7d8","signature":"4ae122398fb2e69f95d7322043d72d18fce83a0a034c8faa5643d673693ae0c2a6ccb049fdff8d2220015f20635f6cb888fec60df2c3ae5eb1e5b6e0e8785cf21c"}
+    
+    // When
+    const metamaskPubKey = signatures.recoverPublicKey(metamask, metamaskPayload, "\u0019Ethereum Signed Message:\n" + JSON.stringify(metamaskPayload).length);
+    const galachainPubKey = signatures.recoverPublicKey(galachain, galachainPayload);
+  
+    // Then
+    console.log('metamaskPubKey', metamaskPubKey, "galachainPubKey", galachainPubKey)
+    expect(metamaskPubKey).toEqual(galachainPubKey)
+  });
 });
```

Comment:
I suggest to rework a little this test, to make it even more clear:

1. Missing `// Given` comment on the top
2. Add `const metamaskPrefix = "\u0019Ethereum Signed Message:\n" + signatures.getPayloadToSign(metamaskPayload).length`. (Note we don't use `JSON.stringify` here, because for some payloads it may produce different input than our way of serializing data)
3. `signature` from `galaChainPayload` may be removed, it's ignored.
4. It would be good to add a comment pointing that `uniqueKey` is different for each payload (this is the only difference, which is important, but also easy to miss)
5. Redundant `console.log`
6. Also please remove redundant quotes from payload field names (is linter not complaining?)